src/Microsoft.OpenApi.CSharpAnnotations.DocumentGeneration/Extensions/StringExtensions.cs

```diff
@@ -105,6 +105,65 @@ public static string RemoveBlankLines(this string value)
                 .Trim();
         }
 
+        /// <summary>
+        /// Removes the duplicate string from the parameter name to work around the roslyn issue.
+        /// https://github.com/dotnet/roslyn/issues/26292.
+        /// </summary>
+        /// <remarks>
+        /// If a character is not allowed in an identifier name in .NET, roslyn duplicates the string following the
+        /// first occurence of that special character in the complied xml.
+        /// e.g if the param name is service-Catalog-Id-1, roslyn will convert it to
+        /// service-Catalog-Id-1-Catalog-Id-1 and this method will remove the duplicate string -Catalog-Id-1 and
+        /// return service-Catalog-Id-1
+        /// Also @ is allowed at the start of the identifier name, so roslyn duplicates the string following the
+        /// second occurence of that special character in the complied xml.
+        /// e.g if the param name is @skip@skip, roslyn will convert it to
+        /// @skip@skip@skip and this method will remove the duplicate string @skip and
+        /// return @skip@skip
+        /// </remarks>
+        /// <param name="value">The param name to update.</param>
+        /// <returns>The updated value.</returns>
+        public static string RemoveRoslynDuplicateString(this string value)
+        {
+            if (string.IsNullOrEmpty(value))
+            {
+                return value;
+            }
+
+            // Only alphanumeric or hyphen characters are allowed in an identifier name.
+            // So look for special character other than these.
+            string firstOccuredSpecialCharacter = Regex.Match(value, "[^a-zA-Z0-9_]").Value;
+
+            if (string.IsNullOrWhiteSpace(firstOccuredSpecialCharacter))
+            {
+                return value;
+            }
+
+            var specialCharIndex = value.IndexOf(firstOccuredSpecialCharacter.ToCharArray()[0]);
+
+            // @ is allowed at the start of the identifier name, so look for second occurence of it.
+            // e.g. if a param name is @skip@skip, roslyn will compile it as @skip@skip@skip.
+            if (firstOccuredSpecialCharacter == "@")
+            {
+                specialCharIndex = value.IndexOf(firstOccuredSpecialCharacter.ToCharArray()[0], 1);
```

Comment:
I don't think this is correct.

If param name is @skip-skip-skip, the desired result is @skip-skip.

You are assuming that the second special character is also an "@" here. The easiest way to solve this might be to just redo the regex match on the substring without the first letter "@".

Reply:
good catch:-).

---
In reply to: [253719948](https://github.com/Microsoft/OpenAPI.NET.CSharpAnnotations/pull/198#discussion_r253719948) [](ancestors = 253719948)